code/modules/forensics/forensic_group.dm

```diff
@@ -0,0 +1,192 @@
+		switch(value_usage)
+			if(FORENSIC_VALUE_IGNORE)
+				return
+			if(FORENSIC_VALUE_SUM)
+				data_old.value += data_new.value
+			if(FORENSIC_VALUE_MULT)
+				data_old.value *= data_new.value
+			if(FORENSIC_VALUE_MIN)
+				data_old.value = min(data_old.value, data_new.value)
+			if(FORENSIC_VALUE_MAX)
+				data_old.value = max(data_old.value, data_new.value)
+
+	copy_group(var/datum/forensic_holder/new_holder, var/include_trace = FALSE)
+		for(var/datum/forensic_data/basic/G in src.evidence_list)
+			if(!HAS_FLAG(G.flags, FORENSIC_TRACE) || include_trace)
+				new_holder.add_evidence(G.get_copy(), src.category)
+
+	remove_evidence(var/datum/forensic_holder/parent, var/removal_flags)
+		if(!HAS_ANY_FLAGS(src.group_flags, removal_flags))
+			return
+		for(var/i in 1 to length(src.evidence_list))
+			if(src.evidence_list[i].should_remove(removal_flags))
+				src.evidence_list.Cut(i, i+1)
+
+/datum/forensic_group/basic_list/sleuth // Used to store smells for Pug sleuthing
+	category = FORENSIC_GROUP_SLEUTH
+	group_flags = FORENSIC_REMOVAL_CLEAN
+
+	get_header()
+		return "Sleuth"
+
+	proc/get_sleuth_text(var/atom/A, var/sleuth_all = FALSE, var/accuracy = -1)
+		if(length(src.evidence_list) == 0)
+			return null
+		if(!sleuth_all)
+			return sleuth_data(A, src.evidence_list[rand(1, length(src.evidence_list))], accuracy, TRUE)
+
+		var/data_text = ""
+		for(var/i in 1 to length(src.evidence_list))
+			data_text += sleuth_data(A, src.evidence_list[i], accuracy, i == 1)
+		return data_text
+
+	proc/sleuth_data(var/atom/A, var/datum/forensic_data/basic/slueth_data, var/accuracy, var/is_first)
+		var/color = slueth_data.evidence.id
+		var/time_since = TIME - slueth_data.time_end
+		var/time = slueth_data.get_time_estimate(accuracy * FORENSIC_BASE_ACCURACY)
+		var/list/time_since_list = list(0, rand(4,6), rand(8,12), rand(27,33), rand(41,49), rand(55,65))
+		var/c_text
+		if(is_first)
+			var/list/intensity_list = list("faintly","acutely","strongly","mildly","kind","trace")
+			var/intensity = get_intensity(intensity_list, time_since_list, time_since)
+			c_text = "\The [A] smells [intensity] of \a [color]."
+		else
+			var/list/intensity_list = list("a faint","an acute","a strong","a mild","kind of a","a trace")
+			var/scent = pick("scent", "hint", "taste", "aroma", "fragrance")
+			var/detect = pick("detect","notice","note","find","pick up","smell","locate","track","discover","acertain","inhale","sense")
+			var/intensity = get_intensity(intensity_list, time_since_list, time_since)
+			c_text = "You also [detect] [intensity] [scent] of [color]."
+		return "<li>[SPAN_NOTICE(c_text)] [time]</li>"
+
+	proc/get_intensity(var/list/intensity_list, var/list/time_since_list, var/time_since)
+		for(var/i in 2 to length(intensity_list))
+			if(time_since < time_since_list[i] MINUTES)
+				return intensity_list[i]
+		return intensity_list[1]
+
+/datum/forensic_group/multi_list // Two or three pieces of evidence grouped together
```

Comment:
Why do we have both this and `/datum/forensic_data/multi`? It seems like you're replicating the same functionality at two different levels which is incredibly confusing.